### PR TITLE
Remove jquery-ui from batch 4 of scripts

### LINKED ISSUE #3009: Remove jquery-ui from batch 4 of scripts
Remove jquery-ui from following scripts:
interface/patient_file/encounter/superbill_custom_full.php
interface/patient_file/front_payment.php
interface/patient_file/merge_patients.php
interface/patient_file/summary/add_edit_issue.php
interface/patient_file/summary/immunizations.php
interface/patient_file/summary/pnotes_full.php
interface/patient_tracker/patient_tracker.php
interface/therapy_groups/therapy_groups_views/header.php
interface/usergroup/facilities.php
interface/usergroup/facility_admin.php

### PR DESCRIPTION
**Pull request by im-Amitto**
Fixes #3009

### CODE REVIEW

File: interface/patient_file/encounter/superbill_custom_full.php [hunk 1]
```diff
@@ -283,7 +283,7 @@ function bucks($amount)
 <head>
     <title><?php echo xlt("Codes"); ?></title>
 
-    <?php Header::setupHeader(['jquery-ui', 'jquery-ui-base']); ?>
+    <?php Header::setupHeader([]); ?>
```
Comment (tywrenn) [inline interface/patient_file/encounter/superbill_custom_full.php]: Remove [] brackets as they aren't needed. Should be just `Header::setupHeader();`.

Comment (tywrenn) [reply interface/patient_file/encounter/superbill_custom_full.php]: Also this file uses JQUI autocomplete which needs removal

Comment (im-Amitto) [reply interface/patient_file/encounter/superbill_custom_full.php]: I am reading your PR to get the better idea of what needs to be changed \ (•◡•) /.

File: interface/patient_file/summary/immunizations.php [hunk 2]
```diff
@@ -323,7 +323,7 @@ function saveImmunizationObservationResults($id, $immunizationdata)
 <html>
```
Comment (tywrenn) [inline interface/patient_file/summary/immunizations.php]: This file still has JQUI autocomplete in it

File: interface/patient_file/front_payment.php [hunk 3]
```diff
@@ -417,7 +417,7 @@ function calcTaxes($row, $amount)
     ?>
 
 <title><?php echo xlt('Receipt for Payment'); ?></title>
-    <?php Header::setupHeader(['jquery-ui']); ?>
+    <?php Header::setupHeader([]); ?>
```
Comment (tywrenn) [inline interface/patient_file/front_payment.php]: Remove [] brackets

File: interface/patient_file/merge_patients.php [hunk 4]
```diff
@@ -33,7 +33,7 @@
 
 <head>
 <title><?php echo xlt('Merge Patients'); ?></title>
-    <?php Header::setupHeader(['jquery-ui']); ?>
+    <?php Header::setupHeader([]); ?>
```
Comment (tywrenn) [inline interface/patient_file/merge_patients.php]: Remove [] brackets

**Comment by tywrenn** [comment]
Done with my review here

File: interface/patient_file/encounter/superbill_custom_full.php [hunk 5]
```diff
@@ -548,7 +555,7 @@ function getCTMask() {
                     <?php if ($mode == "modify") { ?>
                         <input type='text' size='6' name="revenue_code" readonly="readonly" value='<?php echo attr($revenue_code) ?>'>
                     <?php } else { ?>
-                        <input type='text' size='6' class='revcode' name="revenue_code" title='<?php echo xla('Type to search and select revenue code'); ?>' value='<?php echo attr($revenue_code) ?>'>
+                        <select size='6' style='width:150px' class='revcode' name="revenue_code" title='<?php echo xla('Type to search and select revenue code'); ?>' value='<?php echo attr($revenue_code) ?>'>
```
Comment (tywrenn) [inline interface/patient_file/encounter/superbill_custom_full.php]: Need closing `</select>`

File: interface/patient_file/summary/immunizations.php [hunk 6]
```diff
@@ -422,7 +422,7 @@ function saveImmunizationObservationResults($id, $immunizationdata)
             <span class=text>
                 <?php echo xlt('Immunization Lot Number'); ?>            </span>          </td>
           <td>
-            <input class='text auto' type='text' name="lot_number" size="25" value="<?php echo attr($lot_number); ?>">          </td>
+            <select class='text auto' type='text' name="lot_number" style='width: 100px' size="25" value="<?php echo attr($lot_number); ?>">          </td>
```
Comment (tywrenn) [inline interface/patient_file/summary/immunizations.php]: Need closing `</select>`

File: interface/patient_file/summary/immunizations.php [hunk 7]
```diff
@@ -1166,12 +1166,31 @@ function sel_code(id)
...
-    source: "../../../library/ajax/imm_autocomplete/search.php?csrf_token_form=" + <?php echo js_url(CsrfUtils::collectCsrfToken()); ?>,
-    minLength: 1
-  });
+  $(".auto").select2({
+        ajax: {
+            url: "../../../library/ajax/imm_autocomplete/search.php",
+            dataType: 'json',
+            data: function(params) {
+                return {
+                  csrf_token_form: <?php echo js_url(CsrfUtils::collectCsrfToken()); ?>,
```
Comment (bradymiller) [inline interface/patient_file/summary/immunizations.php]: data will be url encoded automatically, so this only needs json encoding now:
```
csrf_token_form: <?php echo js_escape(CsrfUtils::collectCsrfToken()); ?>,
```

**Comment by bradymiller** [comment]
hi @im-Amitto , Testing well and code looks good (there is a super minor issue I posted above which I will happily take liberty of adjusting in a commit following this). Bringing this into codebase. Thanks for the codebase improvements! -brady

**Comment by im-Amitto** [comment]
@bradymiller i acutally found a another bug, should't we allow to add custom lot number which does not appear during search?

**Comment by bradymiller** [comment]
hi @im-Amitto ,
Good catch. Agree will basically make it impossible to ever enter in lot numbers (note it is basically searching prior lot numbers, so i none exist, then no choices). Tested your fix, which works well, and will bring in to keep things moving fast :)

**Comment by bradymiller** [comment]
fix in:
https://github.com/openemr/openemr/commit/6ad41fbd9c32cdb50849d1327e40eb60344b01f7

**Pull request merged**